src/App.tsx

```diff
@@ -1,61 +1,179 @@
 import './App.scss';
+import {
+  FormControl,
+  Grid,
+  InputLabel,
+  MenuItem,
+  Paper,
+  Select,
+  SelectChangeEvent,
+  TextField,
+} from '@mui/material';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { useState } from 'react';
+import Button from '@mui/material/Button';
+import { TodoList } from './components/TodoList/TodoList';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Task } from './types/Task';
+
+const tasksFromServer = todosFromServer.map(todo => {
+  return {
+    ...todo,
+    user: { ...usersFromServer.find(user => user.id === todo.userId) },
+  };
+}) as Task[];
 
 export const App = () => {
+  const storageItemsCount: number
+    = JSON.parse(localStorage.getItem('tasks') || '{}').length;
+  const storageSavedTasks: Task[]
+    = JSON.parse(localStorage.getItem('tasks') || '{}');
+
+  const [tasks, setTasks] = useState(storageItemsCount > 3
+    ? storageSavedTasks
+    : tasksFromServer);
+  const [titleTask, setTitleTask] = useState('');
+  const [taskOwnerName, setTaskOwnerName] = useState('0');
+  const [taskOwner, setTaskOwner] = useState<User | {}>({});
+  const [isTitleEmpty, setIsTitleEmpty] = useState(false);
+  const [isTaskOwnerEmpty, setIsTaskOwnerEmpty] = useState(false);
+  const [isTasksSaved, setIsTasksSaved] = useState(true);
+
+  function selectTaskOwner(event: SelectChangeEvent<string>) {
+    setIsTaskOwnerEmpty(false);
+    setTaskOwnerName(event.target.value);
+    setTaskOwner({
+      ...usersFromServer
+        .find(user => user.name === event.target.value),
+    });
+  }
+
+  function getNextId(addedTasks: Task[]): number {
+    addedTasks.sort((taskOne, taskTwo) => taskTwo.id - taskOne.id);
+
+    return addedTasks[0].id + 1;
+  }
+
+  function addNewTask() {
+    if (titleTask.trim().length !== 0 && Object.keys(taskOwner).length > 0) {
+      const newTask: Task = {
+        id: getNextId(tasks),
+        title: titleTask,
+        completed: false,
+        userId: 'id' in taskOwner ? taskOwner.id : 0,
+        user: {
+          id: 'id' in taskOwner ? taskOwner.id : 0,
+          name: 'name' in taskOwner ? taskOwner.name : '',
+          username: 'username' in taskOwner ? taskOwner.username : '',
+          email: 'email' in taskOwner ? taskOwner.email : '',
+        },
```

Comment:
Maybe, let's use specific function for getting user by id ?

Reply:
Done!